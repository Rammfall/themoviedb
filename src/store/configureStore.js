import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'

import logics from './logics'
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {}

  const logicMiddleware = createLogicMiddleware(logics, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
