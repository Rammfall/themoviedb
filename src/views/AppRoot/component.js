import React from 'react'

import Route from 'Views/AppRoot/components/Route'

import LoginPage from '../pages/Login'
import DashboardPage from '../pages/Dashboard'
import ListsPage from '../pages/Lists'
import WatchlistPage from '../pages/Watchlist'

const AppRootComponent = () => (
  <>
    <Route
      path="/login"
      component={LoginPage}
      exact
    />
     <Route
      path="/"
      privateRoute
      component={DashboardPage}
      exact
     />
    <Route
      path="/lists"
      privateRoute
      component={ListsPage}
      exact
    />
    <Route
      path="/watchlist"
      privateRoute
      component={WatchlistPage}
      exact
    />
  </>
)

export default AppRootComponent
