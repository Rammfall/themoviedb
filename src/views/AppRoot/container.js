import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { isLoggedInSelector } from 'Store/concepts/session/selectors'

import AppRootComponent from './component'

const AppRoot = ({ isLoggedIn }) => <AppRootComponent isLoggedIn={isLoggedIn} />

AppRoot.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired
}

const mapStateToProps = (state) => ({
  isLoggedIn: isLoggedInSelector(state)
})

export default connect(mapStateToProps)(AppRoot)
