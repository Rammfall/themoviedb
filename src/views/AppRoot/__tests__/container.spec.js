import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import AppRoot from '../container'

jest.mock('Store/concepts/session/selectors', () => ({
  isLoggedInSelector: jest.fn(() => true)
}))

describe('AppRoot match snapshot', () => {
  let mockStore

  beforeAll(() => {
    mockStore = configureStore([])()
  })

  it('with default props', () => {
    const container = shallow(<AppRoot store={mockStore} />)
      .dive()
      .dive()

    expect(container).toMatchSnapshot()
  })
})
