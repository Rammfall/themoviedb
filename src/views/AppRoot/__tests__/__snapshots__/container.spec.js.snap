// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRoot match snapshot with default props 1`] = `
<AppRootComponent
  isLoggedIn={true}
/>
`;
